directory/directory.go

```diff
@@ -77,13 +77,15 @@ func NewDirImageSource(dir string) types.ImageSource {
 	return &dirImageSource{dir}
 }
 
-func (s *dirImageSource) GetManifest() ([]byte, string, error) {
+func (s *dirImageSource) GetManifest() (types.ImageManifest, string, error) {
 	manifest, err := ioutil.ReadFile(manifestPath(s.dir))
 	if err != nil {
 		return nil, "", err
 	}
+	// TODO(runcom): we should hide this type from the pkg API
```

Comment:
The `ImageManifest` returned by `dirImageSource` needs to understand the manifest contents just as well as the `dockerImageSource` implementation.

I.e. I think the “download a manifest” and “parse and understand a manifest” code should be separate, with the `ImageSource` abstraction serving as the border.